t3/simulate/rmg_simulator.py

```diff
@@ -0,0 +1,273 @@
+                                         use_chemkin_names=False,
+                                         check_duplicates=False,
+                                         )
+        if self.rmg_model is None:
+            self.logger.error('The RMG adapter did not properly read the rmg input file.')
+
+        # update the rmg object to perform SA if applicable
+        if len(self.observable_list):
+            self.logger.info('Running SA using RMG...')
+            self.observable_species = [species for species in self.rmg_model.reaction_model.core.species
+                                       if species.label in self.observable_list]
+
+            for reaction_system in self.rmg_model.reaction_systems:
+                if isinstance(reaction_system, SimpleReactor):
+                    reaction_system.sensitive_species = self.observable_species
+                    reaction_system.sensitivity_threshold = self.t3['sensitivity']['SA_threshold']
+                    if hasattr(reaction_system, 'Trange') and reaction_system.Trange is not None:
+                        temperature = sum([t.value_si for t in reaction_system.Trange]) / len(reaction_system.Trange)
+                    else:
+                        temperature = reaction_system.T.value_si
+                    reaction_system.sens_conditions['T'] = temperature
+                    if hasattr(reaction_system, 'Prange') and reaction_system.Prange is not None:
+                        pressure = sum([p.value_si for p in reaction_system.Prange]) / len(reaction_system.Prange)
+                    else:
+                        pressure = reaction_system.P.value_si
+                    reaction_system.sens_conditions['P'] = pressure
+                elif isinstance(reaction_system, LiquidReactor):
+                    reaction_system.sensitive_species = self.observable_species
+                    reaction_system.sensitivity_threshold = self.t3['sensitivity']['SA_threshold']
+                    if hasattr(reaction_system, 'Trange') and reaction_system.Trange is not None:
+                        temperature = sum([t.value_si for t in reaction_system.Trange]) / len(reaction_system.Trange)
+                    else:
+                        temperature = reaction_system.T.value_si
+                    reaction_system.sens_conditions['T'] = temperature
+                    if hasattr(reaction_system, 'Vrange') and reaction_system.Vrange is not None:
+                        volume = sum([v for v in reaction_system.Vrange]) / len(reaction_system.Vrange)
+                    else:
+                        volume = reaction_system.V
+                    reaction_system.sens_conditions['V'] = volume
+                else:
+                    raise NotImplementedError(f'RMG SA not implemented for Reactor type {type(reaction_system)}.')
+
+        try:
+            self.logger.info('Simulating a job using RMG...')
+            simulate(self.rmg_model)
```

Comment:
Can we check whether the file exists (raise an error if it doesn't) instead of the try-except?
Anyway, the `self.logger` statement should be outside the try-except block.